lib/bolt/plugin/terraform.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Terraform
+      def name
+        'terraform'
+      end
+
+      def hooks
+        ['lookup_targets']
+      end
+
+      def lookup_targets(opts)
+        state = load_statefile(opts)
+
+        resources = state.fetch('modules', {}).flat_map do |mod|
+          mod.fetch('resources', {}).map do |name, resource|
+            [name, resource.dig('primary', 'attributes')]
+          end
+        end
+
+        regex = Regexp.new(opts['resource_type'])
+
+        resources.select do |name, _resource|
+          name.match?(regex)
+        end.map do |_name, resource|
+          target = { 'uri' => resource[opts['uri']] }
+          if opts.key?('name')
+            target['name'] = resource[opts['name']]
+          end
```

Comment:
Should we allow setting `alias`?

Reply:
See example below, but adding an alias would allow something like:
## inventory
```yaml
version: 2
groups:
  - name: clouds
    target-lookups:
      - plugin: terraform
        dir: /home/cas/working_dir/BOLT-1265-tf-plugin
        resource_type: aws_instance.xenial
        uri: public_ip
        alias: id
    config:
      transport: ssh
      ssh:
        user: ubuntu
        host-key-check: false
        private-key: /home/cas/working_dir/BOLT-1265-tf-plugin/keys/aws_key
```
## example using alias as well as group name
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t clouds
Started on 18.236.102.100...
Started on 34.218.228.134...
Started on 54.218.207.166...
Finished on 34.218.228.134:
  STDOUT:
    ubuntu
Finished on 54.218.207.166:
  STDOUT:
    ubuntu
Finished on 18.236.102.100:
  STDOUT:
    ubuntu
Successful on 3 nodes: 18.236.102.100,34.218.228.134,54.218.207.166
Ran on 3 nodes in 0.62 seconds
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t i-020415bcb7f59e199
Started on 18.236.102.100...
Finished on 18.236.102.100:
  STDOUT:
    ubuntu
Successful on 1 node: 18.236.102.100
Ran on 1 node in 0.58 seconds
```

Reply:
Is that different than setting `name`?

Reply:
I guess not. I forgot/didnt realize that setting `uri` overrides anything you specify with `name`.